client/asset/dcr/dcr.go

```diff
@@ -3506,9 +3508,12 @@ func (dcr *ExchangeWallet) SwapConfirmations(ctx context.Context, coinID, contra
 	defer cancel()
 
 	// Check if we can find the contract onchain without using cfilters.
-	_, confs, spent, err = dcr.lookupTxOutput(ctx, txHash, vout)
+	var spendFlag int8
+	_, confs, spendFlag, err = dcr.lookupTxOutput(ctx, txHash, vout)
 	if err == nil {
-		return confs, spent, nil
+		if spendFlag != -1 {
+			return confs, spendFlag > 0, nil
+		} // else go on to block filters scan
```

Comment:
Without this distinction, all of the below code to go to `lookupTxOutWithBlockFilters` NEVER happened.  Also checked lots of logs for the "SwapConfirmations - cfilters scan did not find" string... nothing.

Reply:
Wouldn't that mean that `lookupTxOutput -> GetTransaction` never returned `asset.CoinNotFoundError`? There does seem to be a clear path to the code below.

Reply:
I mean it never happened when the swap was actually confirmed (and found by GetTransaction).  No spending transaction was ever found by `SwapConfirmations` as a result.  Also note that for you own swap, `GetTransaction` always returns a result even when unconfirmed because it's a wallet txn, hence we never determined spend status of our own contract, always relying on revoke to allow find redemption.